com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/CustomMetricsSink.java

```diff
@@ -0,0 +1,168 @@
+/* Generated by Streams Studio: November 29, 2016 at 3:22:39 PM EST */
+package com.ibm.streamsx.metrics;
+
+
+import java.net.InetAddress;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Logger;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingInput;
+import com.ibm.streams.operator.Tuple;
+import com.ibm.streams.operator.model.InputPortSet;
+import com.ibm.streams.operator.model.InputPortSet.WindowMode;
+import com.ibm.streams.operator.model.InputPortSet.WindowPunctuationInputMode;
+import com.ibm.streams.operator.model.InputPorts;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.Parameter;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+
+@PrimitiveOperator(name="CustomMetricsSink", namespace="com.ibm.streamsx.metrics",
+description="Java Operator CustomMetricsSink")
```

Comment:
Currently, the MetricsSink outputs the metrics to a database that Grafana and Kibana can query from.
Pete suggested that it might be useful for users to output their own custom data (not necessarily metrics) to the database. For example, I connected it to a health application which outputted heart rate and ECG data to the database.
So, I quickly made this operator which simply takes tuple attribute names as a parameter and outputs it to the database. Not sure what the name of the operator should be or whether we should keep it in the metrics toolkit.